scripts/ci/check_compliance.py

```diff
@@ -991,6 +993,34 @@ def run(self):
                 self.failure(f"Error parsing {file}: {ex}")
 
 
+class YAMLLint(ComplianceTest):
+    """
+    YAMLLint
+    """
+    name = "YAMLLint"
+    doc = "Check YAML files with YAMLLint."
+    path_hint = "<git-top>"
+
+    def run(self):
+        config_file = os.path.join(ZEPHYR_BASE, ".yamllint")
+
+        for file in get_files(filter="d"):
+            if Path(file).suffix not in ['.yaml', '.yml']:
+                continue
+
+            yaml_config = config.YamlLintConfig(file=config_file)
+
+            if file.startswith(".github/"):
+                # Tweak few rules for workflow files.
+                yaml_config.rules["line-length"] = False
+                yaml_config.rules["truthy"]["allowed-values"].extend(['on', 'off'])
```

Comment:
should we have multiple `.yamllint` files in relevant subdirs? e.g. one for workflows, one for bindings?

Reply:
Yeah I thought a good bit about how to manage overrides. I'm thinking: if we can get away with only this override we may as well just keep it like this, it's nice and compact. If we start having more per directory configuration though we may as well have separate configs.

Reply:
Well, I added another one, but it's at the root directory so per subdir config override does not work there. I'm open to ideas.